src/types.ts

```diff
@@ -67,6 +67,7 @@ export type AppState = {
   openMenu: "canvas" | "shape" | null;
   lastPointerDownWith: PointerType;
   selectedElementIds: { [id: string]: boolean };
+  SelectedElementOriginalPosition: { x: number; y: number }[] | null;
```

Comment:
Hi, it's not easy to notice, but we'd like to avoid adding app state as much as possible.
You could use a variable in a closure during the drag. Please see how `dragOffsetXY` is used.

Reply:
Thanks, fixed this.